test/cmocka/src/math/arithmetic/base2_logarithm.c

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <math.h>
+#include <cmocka.h>
+
+#include <sof/audio/format.h>
+#include <sof/math/base2log.h>
+#include <sof/common.h>
+#include "log2_tables.h"
+/* 'Error[max] = 0.0000236785999981,THD(-dBc) = -92.5128795787487235' */
+#define CMP_TOLERANCE 0.0000236785999981
+
+static void test_math_arithmetic_base2log_fixed(void **state)
+{
+	(void)state;
+
+	double u;
+	double v;
+	int b_i;
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(log2_lookup_table); i++) {
+		u = ((double)base2vector_table[i]);
+		v = fabs(u);
+		/* GitHub macro Q_CONVERT_FLOAT is inaccurate, so replaced with below */
+		u = (v >= 0.5) ? floor(u + 0.5) : 0.0;
+		b_i = (int)u;
+
+		float y = Q_CONVERT_QTOF(base2_logarithm(b_i), 0);
+		float diff = fabs(log2_lookup_table[i] - (double)y/(1 << 16));
+
+		if (diff > CMP_TOLERANCE) {
```

Comment:
Not just print but fail the test with `assert_int_equal(val, ref)`. The prints get unnoticed by CI system.

Reply:
Sure, I will add it